src/main/java/Duke.java

```diff
@@ -1,10 +1,238 @@
+            System.out.println("\t Nice! I've marked this task as done: ");
+            String output = String.format("%02d. [%s] %s", taskIndex, tasks.get(taskIndex - 1).getStatusIcon(), tasks.get(taskIndex - 1).getDescription());
+            System.out.println("\t" + output);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("\t Invalid task number!");
+        } catch (NumberFormatException e) {
+            System.out.println("\t Please enter the task number instead!");
+        }
+    }
+
+    public static void listTasks(){
+        // Lists out the things under tasks.
+        System.out.println("\t Here are the tasks in your list:");
+        int i = 0;
+        for(Task t : tasks){
+            String output = String.format("%02d. %s", i+1, t);
+            System.out.println("\t" + output);
+            i++;
+        }
+    }
+
+    public static void deleteTask(String taskDescription){
+        // Deletes one of the tasks
+        String[] input = taskDescription.split(" ");        // input = ["Done", taskIndex]
+        int taskIndex = Integer.parseInt(input[1]);
+        try {
+            String taskInformation = tasks.get(taskIndex-1).toString();
+            tasks.remove(taskIndex - 1);
+            System.out.println("\t Noted. I have removed this task. ");
+            System.out.println("\t" + taskInformation);
+            System.out.println("\t You now have " + tasks.size() + "tasks in the list.");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("\t Invalid task number!");
+        } catch (NumberFormatException e) {
+            System.out.println("\t Please enter the task number instead!");
+        }
+    }
+
+    private static void handleInput(String line) throws IllegalCommandWordException {
+        line = line.trim();
+        String[] wordsEntered = line.split(" ");
+        if (!COMMAND_WORDS.contains(wordsEntered[0].toLowerCase())) {
+            throw new IllegalCommandWordException();
+        } else if (line.toLowerCase().equals("list")){
+            listTasks();
+        } else if (line.toLowerCase().contains("done")){
+            markTaskAsDone(line);
+        } else if (line.toLowerCase().contains("delete")) {
+            // Insert code for deleting task
+            deleteTask(line);
+        } else {
+            try {
+                addTask(line);
+            } catch (MissingDeadlineException e) {
+                System.out.print("\tSorry, please indicate a valid deadline!\n");
+            } catch (IndexOutOfBoundsException e) {
+                System.out.print("\tSorry, please check if sufficient inputs are given.\n" +
+                        "\tTODO: details\n" +
+                        "\tDeadline: details, deadline\n" +
+                        "\tEvent: details, time of event\n");
+            } catch (MissingEventTimeException e) {
+                System.out.print("\tSorry, please indicate a valid event time!\n");
```

Comment:
There are some details expressed in this `else` statement that could be abstracted away. How about trying [SLAP](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard)?